projects/control-service/projects/base/src/main/java/com/vmware/taurus/service/diag/methodintercept/DiagnosticsPublisher.java

```diff
@@ -64,12 +64,16 @@ public void accept(DiagnosticsContext context) {
 
       ObjectMapper mapper = new ObjectMapper();
       ObjectNode node = mapper.convertValue(metrics, ObjectNode.class);
-      node.put("@table", "taurus_api_call");
-      node.put("@id", opId);
-      String json = node.toPrettyString();
-      log.trace("Sending telemetry: {}", json);
-      this.telemetryClient.sendAsync(json);
+      if (node != null) {
```

Comment:
How could this be nulll?
Metrics can't ever be null at this point.

Reply:
According to the object mapper java doc, the convertValue method could return a null value even if the argument is not null, hence this check.

Reply:
I don't see that in the docs ? https://fasterxml.github.io/jackson-databind/javadoc/2.8/com/fasterxml/jackson/databind/ObjectMapper.html#convertValue(java.lang.Object,%20java.lang.Class)

Reply:
I agree.. sorry... I looked at some examples of edge cases where either the object was null or was of primitive type, which cannot happen at this part of the code. 

I've reverted this change. 

btw, we are using a newer version of the library and the correct javadoc link is: https://fasterxml.github.io/jackson-databind/javadoc/2.13/com/fasterxml/jackson/databind/ObjectMapper.html#convertValue(java.lang.Object,%20java.lang.Class)